### ISSUE TITLE
OpenAPI or Postman

### ISSUE THREAD

**Issue by rodericj** [crobertsbmw/deckofcards#73]
Have you ever considered providing an OpenAPI/Swagger/Postman implementation for this? If I get some time I'd consider creating a PR for it.

**Comment by crobertsbmw** [comment]
I'm not sure I understand what the difference is between what it currently
is and what a OpenAPI/Swagger/Postman implementation would be? You can hit
all the endpoints using Postman, is there something else that is missing?

**Comment by rodericj** [comment]
An OpenAPI implementation there are tools available that will actually generate clients in any language. This helps in my specific case as creating the client can be quite cumbersome.

**Comment by crobertsbmw** [comment]
Yeah, that would be awesome. If you wanted to create PR, I would be happy
to merge it in. The only thing I am concerned about is keeping it backward
compatible.

**Comment by rodericj** [comment]
I've started a draft here: https://app.swaggerhub.com/apis/Thumbworks/DeckOfCards/1.0.0. This'll allow you to export some large number of clients, including swift, which I'm interested in. The idea with my PR would be that I'd add the specific yaml file to a documents folder somewhere. Happy to collaborate if this is something you'd want quicker.

**Comment by crobertsbmw** [comment]
I'm still a little confused; I apologize, I'm not the most experienced when
it comes to these things. So is this going to exist separately from the
main api? Or am I supposed to change the api to have all of these various
things defined?

**Comment by rodericj** [comment]
I mean it depends how deep you want to go with this. One scenario is that you drop the yaml file in a docs directory like this: https://github.com/jamescmartinez/coffee/tree/master/docs. Next level after this would be to build files like [urls.py](https://github.com/crobertsbmw/deckofcards/blob/master/deck/urls.py) and [view.py](https://github.com/crobertsbmw/deckofcards/blob/master/deck/views.py) using the yaml in the docs folder. 

With just dropping the file into the docs directory, you'd need to now keep 2 things up to date.

**Comment by rodericj** [comment]
I went ahead and auto generated some of a client library that I'd be interested in using: https://github.com/ThumbWorks/DeckOfCards. This then opens up opportunities to use swift package manager which I was interested in using.

**Comment by crobertsbmw** [comment]
This is super interesting. I'll have to dig into it a little more. Thanks
for doing this!

**Comment by rodericj** [comment]
I'd spoken to a colleague about this and it became abundantly clear that I needed to add some usage in the README, so take a look at that to see what this gives you.

In working on this I'd come up with some feedback items on the design of this API. I'd like to propose these in a new PR and/or issue if you're up for it.

**Comment by crobertsbmw** [comment]
Of course, that would be great.
